src/query/sql/src/planner/binder/ddl/table.rs

```diff
@@ -565,6 +567,62 @@ impl Binder {
         Ok(Plan::CreateTable(Box::new(plan)))
     }
 
+    #[async_backtrace::framed]
+    pub(in crate::planner::binder) async fn bind_attach_table(
+        &mut self,
+        stmt: &AttachTableStmt,
+    ) -> Result<Plan> {
+        let (catalog, database, table) =
+            self.normalize_object_identifier_triple(&stmt.catalog, &stmt.database, &stmt.table);
+
+        let mut path = stmt.uri_location.path.clone();
+        if path.ends_with('/') {
+            path.pop();
+        }
+        let mut parts = path.split('/').collect::<Vec<_>>();
+        if parts.len() < 2 {
+            return Err(ErrorCode::BadArguments(format!(
+                "Invalid path: {}",
+                stmt.uri_location
+            )));
+        }
+        let storage_prefix = parts.split_off(parts.len() - 2).join("/");
+        let root = parts.join("/");
+        let root = format!("{}/", root);
+        let mut options = BTreeMap::new();
+        options.insert(OPT_KEY_STORAGE_PREFIX.to_string(), storage_prefix);
```

Comment:
it would be better if there are more docs about the logic of parsing and extracting URI

Reply:
I have add some comments, if it is not clear enough, please ping me again